src/interpreter.rs

```diff
@@ -22,14 +22,18 @@ impl<'a, T> Interpreter<'a, T> where T: FromStr {
     fn eval_token(&mut self, token: &str) {
         if let Ok(literal) = token.parse::<T>() {
             self.push(literal);
+        } else {
+            if let Some(&word) = self.vocabulary.get(token) {
+                word.eval_within(self);
+            }
         }
```

Comment:
Most importantly, you changed `vocabulary` to contain references to `Word`s.

Still confused about the borrow/ref semantics… `&&` apparently denotes a double indirection, but I can write `&word.eval_within(…` on line 27 and it still works…